Mlem/Models/Content/Post Model.swift

```diff
@@ -5,26 +5,35 @@
 //  Created by Eric Andrews on 2023-08-26.
 //
 
+import Dependencies
 import Foundation
 
 /// Internal model to represent a post
 /// Note: this is just the first pass at decoupling the internal models from the API models--to avoid massive merge conflicts and an unreviewably large PR, I've kept the structure practically identical, and will slowly morph it over the course of several PRs. Eventually all of the API types that this model uses will go away and everything downstream of the repositories won't ever know there's an API at all :)
-struct PostModel {
-    let postId: Int
-    let post: APIPost
-    let creator: UserModel
-    let community: CommunityModel
-    var votes: VotesModel
-    let commentCount: Int
-    let unreadCommentCount: Int
-    let saved: Bool
-    let read: Bool
-    let published: Date
-    let updated: Date?
-    let links: [LinkType]
+class PostModel: ContentIdentifiable, ObservableObject {
+    @Dependency(\.hapticManager) var hapticManager
+    @Dependency(\.errorHandler) var errorHandler
+    @Dependency(\.postRepository) var postRepository
+    
+    var postId: Int
+    var post: APIPost
+    var creator: UserModel
+    var community: CommunityModel
+    @Published var votes: VotesModel
+    var commentCount: Int
+    var unreadCommentCount: Int
```

Comment:
Do we want the unread comment count to disappear once you've clicked on a post? `unreadCommentCount` needs to be `@Published` if so